packages/pigeon/lib/objc_generator.dart

```diff
@@ -197,6 +215,15 @@ class ObjcHeaderGenerator extends StructuredGenerator<ObjcOptions> {
       indent,
       dartPackageName: dartPackageName,
     );
+    if (root.requiresOverflowClass) {
```

Comment:
For Objective-C, I think you could actually move this to the end of `writeGeneralUtilities` in the implementation file, because you don't need to reference it anywhere in the header.

You'd need to extract the body of the header `writeDataClass` to a file-level `_writeDataClassDeclaration` so you could use it from the implementation file, but I think that would be straightforward?